domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputHasDenominatorEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,185 @@
+package org.oppia.android.domain.classify.rules.fractioninput
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.Fraction
+import org.oppia.android.app.model.InteractionObject
+import org.oppia.android.domain.classify.RuleClassifier
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [FractionInputHasDenominatorEqualToRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class FractionInputHasDenominatorEqualToRuleClassifierProviderTest {
+  private val WHOLE_NUMBER_123 = createWholeNumber(isNegative = false, value = 123)
+  private val FRACTION_1_OVER_2 = createFraction(isNegative = false, numerator = 1, denominator = 2)
+  private val FRACTION_2_OVER_4 = createFraction(isNegative = false, numerator = 2, denominator = 4)
+  private val NON_NEGATIVE_VALUE_1 = createNonNegativeInt(value = 1)
+  private val NON_NEGATIVE_VALUE_2 = createNonNegativeInt(value = 2)
+
+  @Inject
+  internal lateinit var fractionInputHasDenominatorEqualToRuleClassifierProvider:
+    FractionInputHasDenominatorEqualToRuleClassifierProvider
+
+  private val denominatorIsEqualClassifierProvider: RuleClassifier by lazy {
+    fractionInputHasDenominatorEqualToRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Test
+  fun testDenominatorEquals_wholeNumber123Answer_withSignedInt2Input_bothValuesDoNotMatch() {
+    val inputs = mapOf("x" to NON_NEGATIVE_VALUE_2)
+
+    val matches =
+      denominatorIsEqualClassifierProvider.matches(
+        answer = WHOLE_NUMBER_123,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testDenominatorEquals_fraction2Over4Answer_withSignedInt1Input_bothValuesDoNotMatch() {
+    val inputs = mapOf("x" to NON_NEGATIVE_VALUE_1)
+
+    val matches =
+      denominatorIsEqualClassifierProvider.matches(
+        answer = FRACTION_2_OVER_4,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testDenominatorEquals_fraction2Over4Answer_withSignedInt2Input_bothValuesMatch() {
```

Comment:
Shouldn't this be ``fraction1Over2Answer``?

Reply:
Fixed!